promql/engine.go

```diff
@@ -880,17 +883,20 @@ func (enh *EvalNodeHelper) signatureFunc(on bool, names ...string) func(labels.L
 // the given function with the values computed for each expression at that
 // step.  The return value is the combination into time series of all the
 // function call results.
-func (ev *evaluator) rangeEval(f func([]parser.Value, *EvalNodeHelper) Vector, exprs ...parser.Expr) Matrix {
+func (ev *evaluator) rangeEval(f func([]parser.Value, *EvalNodeHelper) (Vector, storage.Warnings), exprs ...parser.Expr) (Matrix, storage.Warnings) {
```

Comment:
I would expect error as well here :thi

Reply:
Error handling is a bit different in engine code. It panics when it encounters error and recovers in a top-level `Eval` method, hence we don't return errors in here and similar helper functions.